packages/react-core/src/components/Alert/__tests__/Alert.test.tsx

```diff
@@ -1,210 +1,724 @@
+  expect(alertContainer).toHaveAttribute('aria-atomic', 'false');
+});
+
+test('Renders with no timeout by default', () => {
+  jest.useFakeTimers();
+
+  render(<Alert title="Some title">Some alert</Alert>);
+
+  act(() => {
+    jest.advanceTimersByTime(8000);
+  });
+
+  expect(screen.getByText('Some alert')).toBeVisible();
+  jest.useRealTimers();
+});
+
+test('Removes the alert after 8000ms when timeout = true', () => {
+  jest.useFakeTimers();
+
+  render(
+    <Alert title="Some title" timeout>
+      Some alert
+    </Alert>
+  );
+
+  act(() => {
+    jest.advanceTimersByTime(7999);
+  });
+
+  expect(screen.getByText('Some alert')).toBeVisible();
+
+  act(() => {
+    jest.advanceTimersByTime(1);
+  });
+
+  expect(screen.queryByText('Some alert')).not.toBeInTheDocument();
+  jest.useRealTimers();
+});
+
+test('Removes the alert after a custom time when timeout is passed with a number', () => {
+  jest.useFakeTimers();
+
+  render(
+    <Alert title="Some title" timeout={2000}>
+      Some alert
+    </Alert>
+  );
+
+  act(() => {
+    jest.advanceTimersByTime(1999);
+  });
+
+  expect(screen.getByText('Some alert')).toBeVisible();
+
+  act(() => {
+    jest.advanceTimersByTime(1);
   });
+
+  expect(screen.queryByText('Some alert')).not.toBeInTheDocument();
+  jest.useRealTimers();
+});
+
+test('Does not remove the alert on timeout if the user is focused on the alert', async () => {
+  const user = userEvent.setup({
+    advanceTimers: delay => jest.advanceTimersByTime(delay)
+  });
+  jest.useFakeTimers();
+
+  render(
+    <Alert title="Some title" timeout>
+      Some alert
+    </Alert>
+  );
+
+  const alert = screen.getByText('Some alert');
+
+  await user.click(alert);
+
+  act(() => {
+    jest.advanceTimersByTime(8000);
+  });
+
+  expect(screen.getByText('Some alert')).toBeVisible();
+  jest.useRealTimers();
+});
+
+test('Does not remove the alert on timeout if the user is hovered over the alert', async () => {
+  const user = userEvent.setup({
+    advanceTimers: delay => jest.advanceTimersByTime(delay)
+  });
+  jest.useFakeTimers();
+
+  render(
+    <Alert title="Some title" timeout>
+      Some alert
+    </Alert>
+  );
+
+  const alert = screen.getByText('Some alert');
+
+  await user.hover(alert);
+
+  act(() => {
+    jest.advanceTimersByTime(8000);
+  });
+
+  expect(alert).toBeVisible();
+  jest.useRealTimers();
+});
+
+test('Removes the alert after the user removes focus from the alert and 3000ms have passed', async () => {
+  const user = userEvent.setup({
+    advanceTimers: delay => jest.advanceTimersByTime(delay)
+  });
+  jest.useFakeTimers();
+
+  render(
+    <div>
+      <input />
+      <Alert title="Some title" timeout>
+        Some alert
+      </Alert>
+    </div>
+  );
+
+  const alert = screen.getByText('Some alert');
+
+  await user.click(alert);
+
+  act(() => {
+    jest.advanceTimersByTime(8000);
+  });
+
+  await user.click(screen.getByRole('textbox'));
+
```

Comment:
For this and the following test, would it make sense to check that the alert is still there after 8000ms and before clicking/hovering away from the alert? Or would the previous tests that cover that be enough?

Reply:
Yeah I debated that myself, but elected to not have that extra check in this test since that aspect of the behavior is already being covered in the preceding test. Happy to add it in if you don't think the other test is enough coverage of it though.

Reply:
I think the only scenario I can imagine where it might come up is if hovering/focusing an alert no longer prevented dismissal (mainly if something else were added/edited/removed to somehow break that functionality). In which case I believe the test would still pass, but the hover/click wouldn't have prevented the dismissal?

That might be quite the edgy edge case, though, so I'd be fine with these three tests as-is if you feel the same way.